python/device/device.py

```diff
@@ -786,12 +807,21 @@ def dispatch_incoming_message_thread(self, worker_thread_info):
                 ):
                     # We only inspect messages that have `thief/runId` and `thief/serviceInstanceId` set to the expected values
                     cmd = thief[Fields.CMD]
-                    if cmd == Commands.SERVICE_ACK_RESPONSE:
+                    if cmd in [Commands.SERVICE_ACK_RESPONSE, Commands.METHOD_RESPONSE]:
                         # If this is a service_ack response, we put it into `incoming_service_ack_response_queue`
                         # for another thread to handle.
-                        logger.info(
-                            "Received {} message with {}".format(cmd, thief[Fields.SERVICE_ACKS])
-                        )
+                        if Fields.SERVICE_ACK_ID in thief:
+                            logger.info(
+                                "Received {} message with {}".format(
+                                    cmd, thief[Fields.SERVICE_ACK_ID]
+                                )
+                            )
+                        else:
+                            logger.info(
+                                "Received {} message with {}".format(
+                                    cmd, thief[Fields.SERVICE_ACKS]
+                                )
+                            )
```

Comment:
Something like this might be cleaner
```suggestion
                        logger.info(
                            "Received {} message with {}".format(
                                cmd, thief[Fields.SERVICE_ACK_ID] if Fields.SERVICE_ACK_ID in thief else thief[Fields.SERVICE_ACKS]
                            )
                        )
```